cartridges/int_algolia/cartridge/scripts/algolia/steps/sendDeltaExportProducts.js

```diff
@@ -108,44 +152,95 @@ function sendDeltaExportProducts(parameters) {
         // there's a folder with a UUID as a name one level down, we need to open that
         var l3_uuidDir = jobHelper.getFirstChildFolder(l2_tempZipDir); // processing/000001.zip/ebff9c4e-ac8c-4954-8303-8e68ec8b190d/
 
-        // UUID-named folder has a folder called "catalogs" in it, open that
+        // UUID-named folder has "catalogs", "inventory-lists" and "pricebooks" in it, open that
         var l4_catalogsDir = new File(l3_uuidDir, 'catalogs'); // processing/000001.zip/ebff9c4e-ac8c-4954-8303-8e68ec8b190d/catalogs/
+        var l4_inventoryListsDir = new File(l3_uuidDir, 'inventory-lists');
+        var l4_pricebooksDir = new File(l3_uuidDir, 'pricebooks');
 
-        // getting child catalog folders, there can be more than one - folder name is the ID of the catalog
-        var l5_catalogDirList = jobHelper.getChildFolders(l4_catalogsDir);
+        // -------------------- processing inventory XML --------------------
 
-        // processing catalog.xml files in each
-        l5_catalogDirList.forEach(function(l5_catalogDir) {
+        if (l4_inventoryListsDir.exists() && l4_inventoryListsDir.isDirectory()) {
 
-            var catalogFile = new File(l5_catalogDir, 'catalog.xml');
+            let inventoryFileList = jobHelper.getAllXMLFilesInFolder(l4_inventoryListsDir);
 
-            // adding productsIDs from the XML to the list of changed productIDs
-            // success contains the number of successfully read records or false if an error occurred
-            success = jobHelper.updateChangedProductsObjectFromXML(catalogFile, changedProducts);
+            inventoryFileList.forEach(function(inventoryFile) {
+                let result = jobHelper.updateCPObjectFromXML(inventoryFile, changedProducts, 'inventory');
 
-            if (success) {
-                // success contains the number of successfully read records
-                productLogData.processedRecords += success;
+                if (result.success) {
+                    productLogData.processedRecords += result.nrProductsRead;
+                } else {
+                    // abort if error reading from any of the delta export zips
```

Comment:
If we abort without removing the problematic files, it will potentially fail again the next time and all subsequent times, won't it?